### ISSUE TITLE
Non-default `color_cycle` not working in Pie plot

### ISSUE THREAD

**Issue by rubennj** [matplotlib/matplotlib#4975]
Changing the default `color_cycle` works in other charts but not in pie.

```
import matplotlib as mpl
import matplotlib.pylab as plt

mpl.rcParams['axes.color_cycle'] = ['c', 'm', 'y', 'k']

fig, ax = plt.subplots()

ax.pie([15, 30, 45, 10])
```
New colors can be used only changing the `colors` parameter of pie.

> Versions: MPL 1.4.3 - Python 3.4.1 - Windows

**Comment by WeatherGod** [comment]
Yes, the "color cycle" in the pie() function was a quick-n-dirty kludge,
IIRC. The function hard-codes the cycle and doesn't use the internal cycle.
It shouldn't be too difficult to get it hooked up to the new property
cycling feature I am working on right now.

**Comment by languitar** [comment]
:+1:

**Comment by mdboom** [comment]
I'm remilestoning this as 2.0.  Since we're changing the default color cycle for 2.0, we should definitely fix the hard-coded one in `pie`.

**Comment by mdboom** [comment]
This is fixed by #5674.

**Comment by tacaswell** [comment]
closed by #6291
